### PR TITLE
feature/android_studio

### PR DESCRIPTION
**Pull request by brettywhite**
converted hello_sdl_android to Android Studio. Tested working on TDK

### CODE REVIEW

File: hello_sdl_android/app/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (joeygrover) [inline hello_sdl_android/app/.gitignore]: There are 3 gitignore files, we shouldn't need this many. Also since this is a high level item that you want to ignore in all projects it should go to the top level gitignore. It looks like a line was removed, but `build/` in the top level gitignore will ignore all build folders.

File: hello_sdl_android/app/src/androidTest/java/com/sdl/hellosdlandroid/ExampleInstrumentedTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import static org.junit.Assert.*;
+
+/**
+ * Instrumentation test, which will execute on an Android device.
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class ExampleInstrumentedTest {
```
Comment (joeygrover) [inline hello_sdl_android/app/src/androidTest/java/com/sdl/hellosdlandroid/ExampleInstrumentedTest.java]: This is the sample unit test case, it should not be a part of this project.

File: hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/MainActivity.java [hunk 3]
```diff
@@ -19,6 +19,10 @@ protected void onCreate(Bundle savedInstanceState) {
 		setContentView(R.layout.activity_main);
 		//If we are connected to a module we want to start our SdlService
 		SdlReceiver.queryForConnectedService(this);
+
+		// Start the SDLService (IF USING EMULATOR, NOT BT)
+        /*Intent sdlServiceIntent = new Intent(this, SdlService.class);
+        startService(sdlServiceIntent);*/
```
Comment (joeygrover) [inline hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/MainActivity.java]: This change is out of scope of this PR

File: hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlService.java [hunk 4]
```diff
@@ -163,6 +168,8 @@ public void startProxy(boolean forceConnect) {
 			try {
                 Log.i(TAG, "Starting SDL Proxy");
 				proxy = new SdlProxyALM(this, APP_NAME, true, APP_ID,new MultiplexTransportConfig(getBaseContext(), APP_ID));
+				// USE TCP FOR EMULATOR (no BlueTooth)
+				//proxy = new SdlProxyALM(this,APP_NAME, true, APP_ID ,new TCPTransportConfig(CORE_PORT, CORE_IP, false));
```
Comment (joeygrover) [inline hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlService.java]: This change is out of scope of this PR

File: hello_sdl_android/app/src/test/java/com/sdl/hellosdlandroid/ExampleUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+public class ExampleUnitTest {
```
Comment (joeygrover) [inline hello_sdl_android/app/src/test/java/com/sdl/hellosdlandroid/ExampleUnitTest.java]: This should not be commited

File: hello_sdl_android/sdl_android/src/androidTest/assets/json/AddCommand.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{
+  "request":{
+    "name":"AddCommand",
+    "correlationID":105,
+    "parameters":{
```
Comment (joeygrover) [inline hello_sdl_android/sdl_android/src/androidTest/assets/json/AddCommand.json]: None of these json files should be included into the tests project

**Comment by brettywhite** [comment]
@joeygrover the reviewed changes have been made in the last commit. thanks

File: hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlService.java [hunk 7]
```diff
@@ -103,7 +104,7 @@
 	private static final String TAG 					= "SDL Service";
 
 	private static final String APP_NAME 				= "Hello Sdl";
-	private static final String APP_ID 					= "8675309";
+	private static final String APP_ID 					= "86753809";
```
Comment (joeygrover) [inline hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlService.java]: This shouldn't be changed.

File: hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlService.java [hunk 8]
```diff
@@ -88,6 +88,7 @@
 import com.smartdevicelink.proxy.rpc.enums.TextAlignment;
 import com.smartdevicelink.proxy.rpc.listeners.OnRPCResponseListener;
 import com.smartdevicelink.transport.MultiplexTransportConfig;
+import com.smartdevicelink.transport.TCPTransportConfig;
```
Comment (joeygrover) [inline hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlService.java]: Make sure to remove this import

File: hello_sdl_android/sdl_android/src/androidTest/AndroidManifest.xml [hunk 9]
```diff
@@ -0,0 +1,19 @@
+<manifest package="com.smartdevicelink" xmlns:android="http://schemas.android.com/apk/res/android">
+	<uses-sdk android:minSdkVersion="8"/>
+        <uses-permission android:name="android.permission.BLUETOOTH" />
```
Comment (joeygrover) [inline hello_sdl_android/sdl_android/src/androidTest/AndroidManifest.xml]: Library test project is still part of the project.

Comment (brettywhite) [inline hello_sdl_android/sdl_android/src/androidTest/AndroidManifest.xml]: knock on wood that thats it 👍 , see latest commit
